java/DataLake/src/main/java/BookReader.java

```diff
@@ -0,0 +1,23 @@
+import java.util.List;
+
+public interface BookReader {
+    /**
+     * Preprocess the text of the book and return it in a list of words.
+     * @param text The text of the book.
+     * @return A list of words after preprocessing.
+     */
+    List<String> preprocessText(String text);
+
+    /**
+     * Get the path separator based on the current operating system.
+     * @return The path separator (e.g., "/" or "\").
+     */
+    String getPathSeparator();
+
+    /**
+     * Read a book from a given path and return a list containing the book name and its words.
+     * @param path The path to the book file.
+     * @return A list containing the book name and its words.
+     */
+    List<Object> readBook(String path);
```

Comment:
If you say you return a list containing the book name and its word, why do you return a `List` of **`Objects`**?

Reply:
Okey ill do this night